### PR TITLE
brod_cli

### PR DESCRIPTION
**Pull request by zmstone**
Sending the review before adding test cases.

### CODE REVIEW

**Comment by id** [comment]
Awesome stuff! I'll test a bit later today.

File: src/brod_cli.erl [hunk 1]
```diff
@@ -0,0 +1,790 @@
...
+      docopt:docopt(Doc, Args, [debug || LogLevel =:= ?LOG_LEVEL_DEBUG])
+    catch
+      C1 : E1 ->
+        Stack1 = erlang:get_stacktrace(),
+        verbose("~p:~p\n~p\n", [C1, E1, Stack1]),
+        print(Doc),
+        erlang:Stop(?LINE)
+    end,
+  case LogLevel =:= ?LOG_LEVEL_QUIET of
+    false ->
```
Comment (id) [inline src/brod_cli.erl]: swap true/false

File: README.md [hunk 2]
```diff
@@ -355,7 +355,12 @@ brod:fetch(Hosts, Topic, Partition, 1).
 # Self-contained binary (needs erlang runtime)
 This will build a self-contained binary with brod application
 
-```shell
-make escript
-./brod help
+```bash
+BROD_CLIE=true make escript
```
[the remark this answers is not in the document]
Comment (id) [reply README.md]: don't need this if you implement my suggestion below

File: Makefile [hunk 3]
```diff
@@ -17,6 +24,11 @@ CT_OPTS = -ct_use_short_names true
 
 include erlang.mk
 
+ifeq ($(BROD_CLI),true)
```
[the remark this answers is not in the document]
Comment (zmstone) [reply Makefile]: Won't work because it's `TEST_ERLC_OPTS ?= ....` in `erlang.mk`.

[the remark this answers is not in the document]
Comment (id) [reply Makefile]: Well, it does not work for me like this.

File: src/brod_cli_pipe.erl [hunk 4]
```diff
@@ -17,7 +17,7 @@
 %% @doc The input source of brod-cli pipe command
 %% This module implements a process that reads off the bytes
 %% from the data source (either stdin or a file)
-%% and sends the byts to parent process.
+%% and sends the byets to parent process.
```
Comment (id) [inline src/brod_cli_pipe.erl]: nice fix :)

File: src/brod_cli.erl [hunk 5]
```diff
@@ -0,0 +1,952 @@
...
+  , {isr, Isr}
+  , {osr, Osr}
+  ] = Info,
+  io_lib:format("~7s: ~p ~s ~s\n",
+                [integer_to_list(Partition), LeaderNodeId,
+                 format_list(Isr),
+                 format_list(Osr)]);
+format_partition_lines({Partition, ErrorCode}) ->
+  ErrorStr = format_error_code(ErrorCode),
+  io_lib:format("    ~s: [ERROR] ~s\n", [Partition, ErrorStr]).
```
Comment (id) [inline src/brod_cli.erl]: integer_to_list(Partition)

**Review by id — APPROVED** [review]
👍

**Review by id — APPROVED** [review]

**Review by id — APPROVED** [review]

**Comment by zmstone** [comment]
rebased to resolve conflicts

**Review by id — APPROVED** [review]

**Pull request merged**